app/models/permission.rb

```diff
@@ -0,0 +1,7 @@
+class Permission < ApplicationRecord
+  enum :name => {:read => "read", :delete => "delete", :order => "order", :update => "update"}, :_suffix => true
+
+  def readonly?
```

Comment:
@syncrou Where is this being called from?

Reply:
This is setting the model as `readonly` and will not allow any updates on a record.
The `enum` type already locks the values that can be used, but in the case an update is called on a record with a declared `enum` value, this stops that update from happening.
```
2.5.5 :001 > permission = Permission.first
  Permission Load (3.8ms)  SELECT  "permissions".* FROM "permissions" ORDER BY "permissions"."id" ASC LIMIT $1  [["LIMIT", 1]]
 => #<Permission id: 1, created_at: "2020-01-27 22:05:52", updated_at: "2020-01-27 22:05:52", name: "read"> 
2.5.5 :002 > permission.name
 => "read" 
2.5.5 :003 > permission.update(:name => 'update')
   (1.6ms)  BEGIN
   (1.3ms)  ROLLBACK
Traceback (most recent call last):
        1: from (irb):3
ActiveRecord::ReadOnlyRecord (Permission is marked as readonly)
```